### PR TITLE
Add support for Idempotent Requests

### PR DESCRIPTION
**Pull request by joeirimpan**
@prakashpp Please review

### CODE REVIEW

File: transaction.py [hunk 1]
```diff
@@ -73,6 +73,12 @@ def authorize_stripe(self, card_info=None):
         stripe.api_key = self.gateway.stripe_api_key
 
         charge_data = self.get_stripe_charge_data(card_info=card_info)
+        charge_data.update({
+                'idempotency_key': '_'.join([
```
Comment (prakashpp) [inline transaction.py]: charge_data = "auth_%s" % self.uuid

File: transaction.py [hunk 2]
```diff
@@ -140,6 +146,12 @@ def capture_stripe(self, card_info=None):
...
```
Comment (prakashpp) [inline transaction.py]: charge_

File: transaction.py (hunk 1 above)
Comment (joeirimpan) [inline transaction.py]: fixed

**Comment by prakashpp** [comment]
LGTM

**Comment by sharoonthomas** [comment]
I would recommend adding this to the cancel transaction too (though it does not really matter). Once cancelled, it is not going to cancel it again ;)

**Pull request merged**